File: python/a33.py
# A33 Game 2
# submission: https://atcoder.jp/contests/tessoku-book/submissions/38740310

n = int(input())
a = list(map(int, input().split()))

xor_sum = a[0]
for i in range(n - 1):
    xor_sum ^= a[i + 1]

if xor_sum == 0:
    print('Second')
else:
    print('First')

File: rust/tessoku-book/src/bin/a33.rs
// A33 Game 2
// submission: https://atcoder.jp/contests/tessoku-book/submissions/38740381

use proconio::input;

fn main() {
    input!{
        n: usize,
        a:[u32; n],
    }
    let mut xor_sum = a[0];
    for i in 1..n {
        xor_sum ^= a[i];
    }
    if xor_sum != 0 {
        println!("First");
    } else {
        println!("Second");
    }
}
